lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/file/MultipartUploadDAOImpl.java

```diff
@@ -288,16 +285,30 @@ private byte[] extractBytes(String requestBody) {
 			throw new RuntimeException(e);
 		}
 	}
+	
+	private CompositeMultipartUploadStatus getUploadStatus(String idString, boolean withLockNoWait) {
+		ValidateArgument.required(idString, "UploadId");
+		try {
+			long id = Long.parseLong(idString);
+			return getUploadStatus(id, withLockNoWait);
+		} catch (NumberFormatException e) {
+			throw new IllegalArgumentException("Invalid UploadId.");
+		}
+	}
 
 	/**
 	 * Get the upload status given an upload id.
 	 * 
 	 * @param id
 	 * @return
 	 */
-	private CompositeMultipartUploadStatus getUploadStatus(long id) {
+	private CompositeMultipartUploadStatus getUploadStatus(long id, boolean lockNoWait) {
 		try {
-			return this.jdbcTemplate.queryForObject(SELECT_BY_ID, STATUS_MAPPER, id);
+			StringBuilder sql = new StringBuilder(SELECT_BY_ID);
+			if (lockNoWait) {
+				sql.append(" FOR UPDATE NOWAIT");
+			}
+			return this.jdbcTemplate.queryForObject(sql.toString(), STATUS_MAPPER, id);
 		} catch (EmptyResultDataAccessException e) {

```

Comment:
can we catch the exception thrown by spring/mysql when no wait does not return a lock.  Then throw a specific exception (TemporarilyUnavailableException). With an autowire test.